packages/fetchai/connections/ledger/connection.py

```diff
@@ -97,12 +97,12 @@ async def send(self, envelope: "Envelope") -> None:
         :param envelope: the envelope to send.
         :return: None
         """
-        task = await self._schedule_request(envelope)
+        task = self._schedule_request(envelope)
         self.receiving_tasks.append(task)
         self.task_to_request[task] = envelope
         self.event_new_receiving_task.set()
 
-    async def _schedule_request(self, envelope: Envelope) -> Task:
+    def _schedule_request(self, envelope: Envelope) -> Task:
```

Comment:
`_schedule_request` wasn't needed to be an asynchronous method. It doesn't change the behavior because we weren't using `await` anyway. But I think it simplifies the analysis.